main.nf

```diff
@@ -1,277 +1,1016 @@
+                saveAs: { params.save_reference ? it : null }, mode: params.publish_dir_mode
+
+        input:
+        file fasta from fasta_bowtie_index
+
+        output:
+        file "${fasta.baseName}.index*" into bowtie_index
+
+        script:
+        """
+        bowtie-build --threads $task.cpus ${fasta} ${fasta.baseName}.index
+        """
+    }
+}
+
+/*
+ * STEP 3 - Cut enzyme binding site at 5' and linker at 3'
+ */
+
+if(!params.skip_trimming && (params.trim_ecop || params.trim_linker)){
+    process trim_adapters {
+        tag "$name"
+        publishDir "${params.outdir}/trimmed/adapter_trimmed", mode: params.publish_dir_mode,
+                saveAs: {filename ->
+                    if (filename.indexOf(".fastq.gz") == -1)    "logs/$filename"
+                    else if (params.save_trimmed) "$filename"
+                    else null
+                }
+
+        input:
+        set val(name), file(reads) from read_files_trimming
+
+        output:
+        set val(name), file("*.fastq.gz") into trimmed_reads_trim_5g
+        file "*.output.txt" into cutadapt_results
+
+        script:
+        // Cut Both EcoP and Linker
+        if (params.trim_ecop && params.trim_linker){
+            """
+            cutadapt -a ^${params.eco_site}...${params.linker_seq} \\
+                --match-read-wildcards \\
+                --minimum-length 15 --maximum-length 40 \\
+                --discard-untrimmed \\
+                --quality-cutoff 30 \\
+                --cores=$task.cpus \\
+                -o "${name}".adapter_trimmed.fastq.gz \\
+                $reads \\
+                > "${name}"_adapter_trimming.output.txt
+            """
+        }
+
+        // Cut only EcoP site
+        else if (params.trim_ecop && !params.trim_linker){
+            """
+            mkdir trimmed
+            cutadapt -g ^${params.eco_site} \\
+                -e 0 \\
+                --match-read-wildcards \\
+                --minimum-length 20 --maximum-length 40 \\
+                --discard-untrimmed \\
+                --quality-cutoff 30 \\
+                --cores=$task.cpus \\
+                -o "${name}".adapter_trimmed.fastq.gz \\
+                $reads \\
+                > "${name}"_adapter_trimming.output.txt
+            """
+        }
+
+        // Cut only Linker
+        else if (!params.trim_ecop && params.trim_linker){
+            """
+            mkdir trimmed
+            cutadapt -a ${params.linker_seq}\$ \\
+                -e 0 \\
+                --match-read-wildcards \\
+                --minimum-length 20 --maximum-length 40 \\
+                --discard-untrimmed \\
+                --quality-cutoff 30 \\
+                --cores=$task.cpus \\
+                -o "${name}".adapter_trimmed.fastq.gz \\
+                $reads \\
+                > "${name}"_adapter_trimming.output.txt
+            """
+        }
+    }
```

Comment:
It looks like here you have the same command but are only changing the first variable each time.

You _could_ set a script variable with a condition that selections which variable to use, to stop duplication of the cutadapt commands, (although if the script condition would be quite long).

e.g.

```
script
def trim_strategy = (param.1 && param.2) ? "-a <XXX>" : (param.1 && !param.2) ? "-g <XXX>" : (param.1 ) <...>   
"""
            mkdir trimmed
            cutadapt "${trim_strategy}" \\
                -e 0 \\
                --match-read-wildcards \\
                --minimum-length 20 --maximum-length 40 \\
                --discard-untrimmed \\
                --quality-cutoff 30 \\
                --cores=$task.cpus \\
                -o "${name}".adapter_trimmed.fastq.gz \\
                $reads \\
                > "${name}"_adapter_trimming.output.txt
            """
```

Reply:
Unfortunately, I don't think I can use it here (same reason why I can't use TrimGalore: CAGE-seq has a tendency to add a `G` on it's 5`-end, so the last trimming step should always be to remove them and I don't think I can combine these into one.

Reply:
Ah :\, fair enough.